examples/subplots_axes_and_figures/subplots_demo.py

```diff
@@ -143,14 +143,16 @@
 # labels of inner Axes are automatically removed by *sharex* and *sharey*.
 # Still there remains an unused empty space between the subplots.
 #
-# The parameter *gridspec_kw* of `.pyplot.subplots` controls the grid
-# properties (see also `.GridSpec`). For example, we can reduce the height
-# between vertical subplots using ``gridspec_kw={'hspace': 0}``.
+# To precisely control the positioning of the subplots, one can explicitly
+# create a `.GridSpec` with `.add_gridspec`, and then call its
+# `~.GridSpec.subplots` method.  For example, we can reduce the height
+# between vertical subplots using ``add_gridspec(hspace=0)``.
 #
 # `.label_outer` is a handy method to remove labels and ticks from subplots
 # that are not at the edge of the grid.
 
-fig, axs = plt.subplots(3, sharex=True, sharey=True, gridspec_kw={'hspace': 0})
+fig = plt.figure()
+axs = fig.add_gridspec(3, hspace=0).subplots(sharex=True, sharey=True)
```

Comment:
I don't like chained commands like this, so maybe the first time you do it:

```
gs = fig.add_gridspec(3, hspace=0)
axes = gs.subplots(shares=True, sharey=True)
```
and then devolve to the "short cut".  With the one-liner I can't help but think that `subplots` is a method on `fig`, whereas its really a method on the `GridSpec`

Reply:
sure, either way is fine for me.